src/ml_board/frontend/newDashboard/src/app/AppNew.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import Home from '../components/Home';
+import DedicatedWorkerClass from '../webworkers/DedicatedWorkerClass';
+import { connect } from 'react-redux';
+import { saveEvalResultData } from '../redux/actions/ExperimentActions';
+
+type AppProps = {
+    evalResult: any
+    saveEvalResultData: Function
+}
+
+type AppState = {
+    
+}
+
+type AppInterface = {
+    mlgymWorker: any
+}
+
+class AppNew extends Component<AppProps, AppState> implements AppInterface{
```

Comment:
-  As per life cycle of react componentDidMount will execute first & will execute only once regardless of the props / redux updates. [in functional components, as any of the props / redux updates, the whole functional component is refreshed again].

- if we use functional component, then the page refreshes again and again due to constant redux updates. So it will cause to create new worker & socket connection everytime forever - going into infinte loop, & thus will crash the app.

- As per life cycle of react componentDidMount will execute first & will execute only once regardless of the props / redux updates. [in functional components, as any of the props / redux updates, the whole functional component is refreshed again].

- If we use functional component, then the page refreshes again and again due to constant redux updates. So it will cause to create new worker & socket connection everytime forever - going into infinte loop, & thus will crash the app.

- So, here we use the class component - which follows the react lifecycle & limiting the page refresh to just once i.e. on screen load / reload.so, here we use the class component - which follows the react lifecycle & limiting the page refresh to just once i.e. on screen load / reload.

Reply:
ah cool! Thanks for the clarification. :)